test/files/run/enrich-gentraversable.scala

```diff
@@ -33,38 +39,23 @@ object Test extends App {
     val fms2 = s.filterMap(c =>if(c % 2 == 0) Some(c.toInt) else None)
     typed[IndexedSeq[Int]](fms2)
     println(fms2)
-  }
-  def testIterable = {
-    class FilterMapImpl[A, Repr](r: Repr, it: IterableOps[A, Iterable, _]) {
-      final def filterMap[B, That](f: A => Option[B])(implicit bf: BuildFrom[Repr, B, That]): That = {
-        val b = bf.newBuilder(r)
-        for(e <- it) f(e) foreach (b +=)
-        b.result
-      }
-    }
-    implicit def filterMap[Repr, A](r: Repr)(implicit fr: IsIterableLike[Repr]): FilterMapImpl[fr.A,Repr] =
-      new FilterMapImpl[fr.A, Repr](r, fr.conversion(r))
 
-    val l = List(1, 2, 3, 4, 5)
-    val fml = l.filterMap(i => if(i % 2 == 0) Some(i) else None)
-    typed[List[Int]](fml)
-    println(fml)
+    val m = Map(1 -> "foo", 2 -> "bar")
+    val fmm = m.filterMap { case (k, v) => if (k % 2 == 0) Some(v -> k) else None }
+    typed[Map[String, Int]](fmm)
+    println(fmm)
 
-    val a = Array(1, 2, 3, 4, 5)
-    val fma = a.filterMap(i => if(i % 2 == 0) Some(i) else None)
-    typed[Array[Int]](fma)
-    println(fma.deep)
+    val tm = TreeMap(1 -> "foo", 2 -> "bar")
+    val tmm = tm.filterMap { case (k, v) => if (k % 2 == 0) Some(v -> k) else None }
+    typed[TreeMap[String, Int]](tmm)
+    println(tmm)
 
-    val s = "Hello World"
-    val fms1 = s.filterMap(c => if(c >= 'A' && c <= 'Z') Some(c) else None)
-    typed[String](fms1)
-    println(fms1)
+    val mv = m.view
+    val fmmv = mv.filterMap { case (k, v) => if (k % 2 == 0) Some(v -> k) else None }
+    typed[View[(String, Int)]](fmmv)
+    println(fmmv.toMap)
 
-    val fms2 = s.filterMap(c =>if(c % 2 == 0) Some(c.toInt) else None)
-    typed[IndexedSeq[Int]](fms2)
-    println(fms2)
   }
 
   testIterableOps
-  testIterable
```

Comment:
I removed the `testIterable` part which was just duplicating what we had in `testIterableOps`. And I added more test cases to the `testIterableOps` (to test with Views and Maps).